### PR TITLE
Add defensive code and build on pre-commit.

### PR DESCRIPTION
**Pull request by mathewjordan**
- Add defensive returns on undefined arguements.
- Be defensive about manifest too.
- Test pre-commit build.

### CODE REVIEW

File: lib/utils/array-helpers.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 export const sample = (array) => {
```
Comment (adamjarling) [inline lib/utils/array-helpers.js:1]: Mind renaming `array` as a parameter to something like `arr`?  Every time I see it I get tripped up that it's a native type:)

File: package.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "export": "next build && next export",
     "lint": "next lint",
     "ts-lint": "tsc --noEmit --incremental --watch",
+    "ts-lint-commit-hook": "tsc --noEmit",
```
Comment (adamjarling) [inline package.json:10]: Nice

File: .husky/pre-commit [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-#npm run ts-lint-commit-hook
-npm run lint && npm run test:ci
\ No newline at end of file
+npm run ts-lint-commit-hook
+npm run lint && npm run test:ci
+npm run build
```
[the remark this answers is not in the document]
Comment (mathewjordan) [reply .husky/pre-commit:6]: For me, this seems like the fullproof way to make sure builds can happen prior to deploying to amplify. We were getting issues on build related `shuffle`, but that shuffle issue was only occurring on a handful of works.

[the remark this answers is not in the document]
Comment (adamjarling) [reply .husky/pre-commit:6]: Ah, yea I suppose in the context of statically generating pages, this is a foolproof way to test.

**Review by adamjarling — APPROVED** [review]
Looking good.

**Pull request merged**